Refactor stakableData/unstakable logic and improve state messages

S0naliThakur opened the pull request:
### **PR Type**
Enhancement


___

### **Description**
- Refactored stakeable and unstakable state logic for clarity

- Improved state messages for staking and unstaking actions

- Added explicit handling for nodes not yet staked

- Reordered node info fetching for better error handling


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>node-commands.ts</strong><dd><code>Refactor stakeable/unstakable logic and improve state messages</code></dd></summary>
<hr>

src/node-commands.ts

<li>Refactored logic to set <code>stakeable</code> and <code>unstakable</code> states based on <br>account status<br> <li> Added clear messages for nodes that have not been staked<br> <li> Moved node info fetching after stakeable logic for improved error <br>handling<br> <li> Ensured stakeable/unstakable objects always have meaningful default <br>values


</details>


  </td>
  <td><a href="https://github.com/shardeum/shardeum-validator-cli/pull/68/files#diff-16e6e49d3bbc5c66140d8f145f7990ca668256de32ecb0a0da81774b14d0b889">+27/-13</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

S0naliThakur committed:
Refactor stakableData/unstakable logic and improve state messages

aniketdivekar commented on src/node-commands.ts:
```diff
@@ -517,6 +525,12 @@ export function registerNodeCommands(program: Command) {
+          } else {
+            unstakable = {
+              unlocked: true,
+              reason: 'Node has not been staked yet. Unstaking is allowed.',
```
shouldn't it be. `Node has not been staked yet. Unstaking is not possible!`

S0naliThakur replied:
Thanks for pointing out. Fixed in new commit

S0naliThakur committed:
Update unstakable reason message to clarify that unstaking is not possible for unstaked nodes

Update unstakable state to reflect that unstaking is not possible for unstaked nodes
